packages/medusa/src/services/order-edit.ts

```diff
@@ -382,4 +383,27 @@ export default class OrderEditService extends TransactionBaseService {
 
     return orderEdit
   }
+
+  async requestConfirmation(
+    orderEditId: string,
+    userId: string
+  ): Promise<OrderEdit> {
+    return await this.atomicPhase_(async (manager) => {
+      const orderEditRepo = manager.getCustomRepository(
+        this.orderEditRepository_
+      )
+      const orderEdit = await this.retrieve(orderEditId)
+
```

Comment:
todo: make this idempotent so that if `requested_at !== null` we just return the orderEdit.

Reply:
should we still emit the `REQUESTED` event or do we not allow re-requests to go out to customers? (would it make sense to allow re-requests after a decline?)

Reply:
Nope we can assume that it has been emitted previously so shouldn't duplicate it :+1: